api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1921,34 +1921,152 @@ type AllocationPool struct {
 	End string `json:"end"`
 }
 
-// RouterParam specifies an OpenStack router to use. It may be specified by either ID.
+// RouterParam specifies an OpenStack router to use. It may be specified by either ID or filter, but not both.
 // +kubebuilder:validation:MaxProperties:=1
 // +kubebuilder:validation:MinProperties:=1
 type RouterParam struct {
 	// ID is the ID of the router to use. If ID is provided, the other filters cannot be provided. Must be in UUID format.
 	// +kubebuilder:validation:Format:=uuid
 	// +optional
 	ID *string `json:"id,omitempty"`
+
+	// Filter specifies a filter to select an OpenStack router. If provided, cannot be empty.
+	Filter *RouterFilter `json:"filter,omitempty"`
+}
+
+// RouterFilter specifies a query to select an OpenStack router. At least one property must be set.
+// +kubebuilder:validation:MinProperties:=1
```

Comment:
We already have MinProperties here, so we should already be rejecting empty filters. I wonder if the inline embedded FilterByNeutronTags messes it up 🤔 Lets leave it out anyway until we discover a need for it.